### PR TITLE
fix: Delete button with no loader in helm app

### PR DESCRIPTION
**Pull request by milindk8**
# Description

Added loader when user delete helm app to prevent multiple call when user clicks delete button multiple times

Fixes [#AB4917](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/4917)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Create and delete helm app & click on delete again 

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx [hunk 1]
```diff
@@ -578,12 +578,12 @@ function ChartValuesView({
         if (commonState.isDeleteInProgress) {
             return
         }
+        dispatch({
+            type: ChartValuesViewActionTypes.isDeleteInProgress,
+            payload: true,
+        })
         getDeleteApplicationApi(deleteAction)
             .then((response: ResponseType) => {
-                dispatch({
```
Comment (vivek-devtron) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:583]: IS it not required?

Comment (milindk8) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:583]: just moved outside

File: src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx [hunk 2]
```diff
@@ -392,6 +393,7 @@ export const DeleteChartDialog = ({
     }
     return (
         <DeleteDialog
+            apiCallInProgress={diableButton}
```
Comment (sohel-rp) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.component.tsx:396]: typo

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:583]: Where are we making it false?

File: src/components/v2/values/chartValuesDiff/ChartValuesView.type.ts [hunk 3]
```diff
@@ -405,6 +405,7 @@ export interface DeleteChartDialogProps {
     handleDelete: (deleteAction: DELETE_ACTION) => void
     toggleConfirmation: (isDeleteConfirmation:boolean) => void
     isCreateValueView?: boolean
+    diableButton?: boolean
```
Comment (sohel-rp) [inline src/components/v2/values/chartValuesDiff/ChartValuesView.type.ts:408]: same here

File: src/components/v2/values/chartValuesDiff/ChartValuesView.tsx (hunk 1 above)
Comment (milindk8) [reply src/components/v2/values/chartValuesDiff/ChartValuesView.tsx:583]: inside the response scope

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**